### PR TITLE
157 sidebar together

### PR DESCRIPTION
**Pull request by brodly**
- Fixed menu sliding bug
- Updated map starting position
- Updated map pins to adhere to backend data shape
- Added styling to Icon Component
- Fixed off-centered checkbox in NC Selector
- Added submit button to call data from backend
- Added redux saga
- Created temp error display for debugging

### CODE REVIEW

File: src/redux/rootSaga.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+function* getData() {
+  const {
+    startDate,
+    endDate,
+    councils,
+    requestTypes,
+  } = yield select(getState, 'data');
+
+  const endPoint = 'http://ec2-3-84-149-45.compute-1.amazonaws.com/pins';
```
Comment (sellnat77) [inline src/redux/rootSaga.js]: AAAAAHHHHHHHHHHHHH
can we move this to an environment variable pllzzzz
#305

Comment (brodly) [reply src/redux/rootSaga.js]: fixed

**Comment by brodly** [comment]
> hey, is `src/styles/styles.css` doing anything? I didn't see it imported anywhere.

I believe it is a leftover from me attempting to fix the funky sidebar issue.

File: src/redux/reducers/data.js [hunk 2]
```diff
@@ -103,13 +122,32 @@ export default (state = initialState, action) => {
...
+        ...state,
+        isLoading: true,
+      };
+    case types.GET_DATA_SUCCESS:
+      return {
+        ...state,
+        data: action.payload,
+        error: null,
+        isLoading: false,
+        lastUpdated: new Date(),
```
[the remark this answers is not in the document]
Comment (brodly) [reply src/redux/reducers/data.js]: Yes, this was a place holder to test. The backend just sends the string `NOW` for now

**Comment by brodly** [comment]
This became a larger PR as I fixed a bug where our linter wasnt working properly and had to resolve a lot of linting issues

**Pull request merged**